src/Proto.Cluster/PubSub/ProducerBatchMessage.cs

```diff
@@ -0,0 +1,67 @@
+// -----------------------------------------------------------------------
+// <copyright file="Messages.cs" company="Asynkron AB">
+//      Copyright (C) 2015-2021 Asynkron AB All rights reserved
+// </copyright>
+// -----------------------------------------------------------------------
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Proto.Remote;
+
+namespace Proto.Cluster
+{
+    public class ProducerBatchMessage :  IRootSerializable
+    {
+        public List<object> Envelopes { get; } = new ();
+
+        public List<TaskCompletionSource<bool>> DeliveryReports { get; } = new();
+
+        public IRootSerialized Serialize(ActorSystem system)
```

Comment:
this showcases the new surrogate serialization. translating to and back from in-mem vs on-the-wire messages